sqlmesh/core/config/connection.py

```diff
@@ -1276,3 +1277,14 @@ def _connection_config_validator(
     mode="before",
     check_fields=False,
 )(_connection_config_validator)
+
+
+if t.TYPE_CHECKING:
+    SerializableConnectionConfig: t.TypeAlias = ConnectionConfig  # type: ignore
+elif PYDANTIC_MAJOR_VERSION >= 2:
+    import pydantic
+
+    # Workaround for https://docs.pydantic.dev/latest/concepts/serialization/#serializing-with-duck-typing
+    SerializableConnectionConfig = pydantic.SerializeAsAny[ConnectionConfig]  # type: ignore
+else:
+    SerializableConnectionConfig = ConnectionConfig  # type: ignore
```

Comment:
If the branch order isn't important, perhaps we could merge the first and the last one?

Reply:
So this part:
```
...: t.TypeAias
```
is crucial for mypy to work correctly, but `TypeAlias` itself has not been introduced until Python 3.10, which means that it'd fail at runtime if evaluated.

Reply:
Will add a comment